src/Sema.zig

```diff
@@ -27678,8 +27704,10 @@ fn cmpNumeric(
             else => return Air.Inst.Ref.bool_false,
         };
         if (lhs_val.isInf()) switch (op) {
-            .gt, .neq => return Air.Inst.Ref.bool_true,
-            .lt, .lte, .eq, .gte => return Air.Inst.Ref.bool_false,
+            .neq => return Air.Inst.Ref.bool_true,
```

Comment:
I think this is incorrect for `inf == inf` and `-inf == -inf` (both should be true)

Reply:
this place is care about float vs int only...
float vs float is handled above.
we need more optimizing for it?

the test below has now passed
```zig
test "inf == inf" {
    const inf = comptime std.math.inf(f64);
    const inf2 = comptime std.math.inf(f64);
    const infr = std.math.inf(f64);
    const infr2 = std.math.inf(f64);
    const neginf = comptime -std.math.inf(f64);
    const neginf2 = comptime -std.math.inf(f64);
    const neginfr = -std.math.inf(f64);
    const neginfr2 = -std.math.inf(f64);
    try std.testing.expect(inf == inf2);
    try std.testing.expect(inf == infr);
    try std.testing.expect(infr == infr2);
    try std.testing.expect(neginf == neginf2);
    try std.testing.expect(neginf == neginfr);
    try std.testing.expect(neginfr == neginfr2);
}
```

Reply:
My mistake, I didn't realize this was mixed int/float

Sorry about that, this looks good :+1: